eng/common/pipelines/templates/jobs/eng-common-workflow-enforcer.yml

```diff
@@ -0,0 +1,27 @@
+trigger: none
+
+pr:
+  branches:
+    include:
+      - master
+  paths:
+    include:
+      - eng/common
+
+jobs:
+  - job: EnforceCommonWorkflow
+
+    pool:
+      vmImage: windows-2019
+
+    steps:
+      - pwsh: |
+          $PrSourceBranch = $(System.PullRequest.SourceBranch)
```

Comment:
If this isn't a PR this step will fail in a weird way so I suggest conditioning it on the whether or not is a PR.